packages/salesforce-adapter/src/adapter.ts

```diff
@@ -583,6 +584,9 @@ export default class SalesforceAdapter implements AdapterOperations {
         client: this.client,
         metadataQuery: buildFilePropsMetadataQuery(baseQuery),
       })
+    const withChangesDetection = (await changedAtOutdated(this.elementsSource))
+      ? false
+      : changesDetectionRequested
```

Comment:
```suggestion
    const withChangesDetection = changesDetectionRequested && await changedAtOutdated(this.elementsSource)
```
That way we avoid invoking `changedAtOutdated` on full fetch.

Reply:
Done (just had to add a !).